design-book/src/design-constraints.md

```diff
@@ -0,0 +1,53 @@
+# Design Constraints
+
+To help ensure that we're building an editor that's both useful to end users and feasible to maintain,
+we've developed a set of design constraints and desiderata. Please try to follow them when making and reviewing changes!
```

Comment:
Nit: I've never heard the term 'desiderata' and googling it just seems to pull up a poem- it might be easier to understand with this removed.

Reply:
```suggestion
we've developed a set of design constraints and goals. Please try to follow them when making and reviewing changes!
```

It's a math / academia term. But agreed, too obscure.